fk/luae_na/src/main/java/se/inera/intyg/common/luae_na/v1/model/converter/certificate/question/QuestionDiagnoser.java

```diff
@@ -0,0 +1,156 @@
+                        Arrays.asList(
+                            DiagnosesListItem.builder()
+                                .id("1")
+                                .build(),
+                            DiagnosesListItem.builder()
+                                .id("2")
+                                .build(),
+                            DiagnosesListItem.builder()
+                                .id("3")
+                                .build()
+                        )
+                    )
+                    .build()
+            )
+            .value(
+                CertificateDataValueDiagnosisList.builder()
+                    .list(createDiagnosValue(value))
+                    .build()
+            )
+            .validation(
+                new CertificateDataValidation[]{
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(DIAGNOS_SVAR_ID_6)
+                        .expression(singleExpression("1"))
+                        .build(),
+                    CertificateDataValidationText.builder()
+                        .limit(LIMIT_DIAGNOSIS_DESC)
+                        .build()
+                }
+            )
+            .build();
+    }
+
+    private static List<CertificateDataValueDiagnosis> createDiagnosValue(List<Diagnos> diagnoses) {
+        if (diagnoses == null) {
+            return Collections.emptyList();
+        }
+
+        final List<CertificateDataValueDiagnosis> newDiagnoses = new ArrayList<>();
+        for (int i = 0; i < diagnoses.size(); i++) {
+            final var diagnosis = diagnoses.get(i);
+            if (diagnosis.getDiagnosKod() == null) {
+                continue;
+            }
+
+            newDiagnoses.add(createDiagnosis(Integer.toString(i + 1), diagnosis));
+        }
+
+        return newDiagnoses;
+    }
+
+    private static CertificateDataValueDiagnosis createDiagnosis(String id, Diagnos diagnos) {
+        return CertificateDataValueDiagnosis.builder()
+            .id(id)
+            .terminology(diagnos.getDiagnosKodSystem())
+            .code(diagnos.getDiagnosKod())
+            .description(diagnos.getDiagnosBeskrivning())
+            .build();
+    }
+
+    public static List<Diagnos> toInternal(Certificate certificate, WebcertModuleService moduleService) {
+        var diagnosisList = diagnosisListValue(certificate.getData(), RespConstants.DIAGNOS_SVAR_ID_6);
+        List<Diagnos> newDiagnosisList = new ArrayList<>();
+        diagnosisList.forEach(diagnosis -> {
+            var newDiagnosis = Diagnos.create(
+                diagnosis.getCode(),
+                diagnosis.getTerminology(),
+                diagnosis.getDescription(),
+                moduleService.getDescriptionFromDiagnosKod(diagnosis.getCode(), diagnosis.getTerminology()));
+            var reuiredListSize = Integer.parseInt(diagnosis.getId()) - 1;
+            while (reuiredListSize >= newDiagnosisList.size()) {
```

Comment:
Should this be part of each forEach-loop? Maybe should be moved outside the loop? It is also a little bit tricky to understand the logic... maybe could be simplified as well?

Reply:
I admit it a bit tricky to understand. The reason i chose this implementation was that it had already been used in:
se.inera.intyg.common.ag7804.v1.model.converter.CertificateToInternal.java
se.inera.intyg.common.lisjp.v1.model.converter.CertificateToInternal.java
I think it could and should be simplified :)

Reply:
I have committed a simplification that keeps the logic separated and should make it easier to read.